Makefile

```diff
@@ -1,58 +1,25 @@
 # Copyright 2014 Nico Reißmann <nico.reissmann@gmail.com>
 # See COPYING for terms of redistribution.
 
-define HELP_TEXT
-clear
-echo "Makefile for the JLM compiler"
-echo "Version 1.1 - 2019-11-26"
-endef
-.PHONY: help
-help:
-	@$(HELP_TEXT)
-	@$(HELP_TEXT_JLM)
-	@echo ""
-	@echo "all                    Compile jlm in release mode, and run unit and C tests"
-	@echo "release                Alias for jlm-release"
-	@echo "debug                  Alias for jlm-debug and check"
-	@echo "docs                   Generate doxygen documentation."
-	@echo "clean                  Alias for jlm-clean"
+LLVMCONFIG = llvm-config-15
 
-JLM_ROOT ?= .
+CPPFLAGS = -I. -Itests $(shell $(LLVMCONFIG) --cflags)
 
-include $(JLM_ROOT)/Makefile.sub
```

Comment:
The Makefile.sub was constructed such that it could be included from the jlm-eval-suite
https://github.com/phate/jlm-eval-suite
such that one could be in the root of jlm-eval-suite and basically perform all the compilation of jlm from there.

For example, jlm-eval-suite has the option to compile the CIRCT project and then compile jlm with the HLS backend enabled. This has been achieved by when standing in jlm-eval-suite and the build of CIRCT is found then the CIRCT_PATH is set, which cause -DCIRCT to be set when building jlm.

How can this be achieved with the new setup?

Reply:
then I would propose to do it this way:
- have a separate "target selection mechanism" (could be "configure"-like script, but can also have default target; could also have "make" targets that switch current config)
- use "build-<config>" directories; e.g. build-debug, build-release, build-hls-release
so you would call to set up build config, then build (and when not changing build config, every "make" will just use presently set config). Dependency tracking etc all works relative to currently set build target. (Most build systems actually handle build different targets in this way).

Reply:
That sounds good, with the added benefit that one can have multiple versions in parallel without having to rebuild everything.

How would we handle the final executables that currently are placed in jlm/bin?
Do we then also have multiple versions placed here, e.g., jlc-release, jlc-debug, and then a way of switching between which one is default (e.g., jlc symlink to either of them)?